src/core/caniuse.js

```diff
@@ -0,0 +1,212 @@
+  if (!conf.caniuse) {
+    return;
+  }
+  // Opportunistically insert the style into the head to reduce FOUC.
+  const codeStyle = document.createElement("style");
+  codeStyle.textContent = caniuseCss;
+  document.head.appendChild(codeStyle);
+
+  normalizeConf(conf);
+  const { caniuse } = conf;
+  const cache = await new IDBCache("respec-caniuse", ["caniuse"]);
+  if (caniuse.feature) {
+    canIUse(
+      caniuse.feature,
+      document.querySelector(".head dl").lastChild,
+      cache,
+      caniuse
+    );
+  }
+}
+
+/**
+ * MUTATES `conf.caniuse` object to hold normalized configurarion
+ * @param {Object} conf   configuration settings
+ */
+function normalizeConf(conf) {
+  const DEFAULTS = {
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours (in ms)
+    browsers: ["chrome", "firefox", "safari", "edge"],
+    versions: 4,
+  };
+
+  if (typeof conf.caniuse === "string") {
+    conf.caniuse = { feature: conf.caniuse, ...DEFAULTS };
+    return;
+  }
+  if (Array.isArray(conf.caniuse.browsers)) {
+    conf.caniuse.browsers = conf.caniuse.browsers.map(b => b.toLowerCase());
+  } else if (conf.caniuse.browsers !== "ALL") {
+    conf.caniuse.browsers = DEFAULTS.browsers;
+  }
+  if (conf.caniuse.maxAge === undefined) conf.caniuse.maxAge = DEFAULTS.maxAge;
+  if (!conf.caniuse.versions) conf.caniuse.versions = DEFAULTS.versions;
+}
+
+/**
+ * main canIUse function
+ * @param  {String} key              which api to look for
+ * @param  {Node} refNode            adds table after refNode
+ * @param {IDBCache} cache           IDB cache
+ * @param {Object} conf              normalized respecConfig.caniuse
+ */
+async function canIUse(key, refNode, cache, conf) {
+  const title = hyperHTML`
+    <dt class="caniuse-title" id="${`caniuse-${key}`}">
+      Can I Use this API? (${key})
+    </dt>`;
+
+  const contentPromise = new Promise(async (resolve) => {
+    // use data from cache data if valid and render
+    try {
+      const cachedStats = await cache.get(key);
+      if (cachedStats
+        && new Date() - new Date(cachedStats.cacheTime) < conf.maxAge) {
+        resolve(getTableHtml(key, cachedStats.stats, conf));
+      }
+    } catch (err) {
+      console.error(err);
+    }
+
+    // otherwise fetch new data, cache and render
+    const url = conf.apiURL
+      ? conf.apiURL.replace("{FEATURE}", key)
+      : `https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/${key}.json`;
+
+    try {
+      const json = await getJson(url);
+      cache.set(key, { stats: json.stats, cacheTime: new Date() })
+        .catch(err => console.error("Failed to cache caniuse data.", err));
+      resolve(getTableHtml(key, json.stats, conf));
+    } catch (err) {
+      const permalink = `http://caniuse.com/#feat=${key}`;
+      resolve(hyperHTML`
+        Error [core/caniuse]: ${err.message}.
+        <br>Please check directly on <a href="${permalink}">${permalink}</a>.`);
+    }
+  });
+
+  const content = hyperHTML`
```

Comment:
The `title` should just be inside here:
```JS
  const content = hyperHTML`
    <dt class="caniuse-title" id="${`caniuse-${key}`}">
      Can I Use this API? (${key})
    </dt>
    <dd class="caniuse-stats">${{
      any: contentPromise,
      placeholder: "Fetching data from caniuse.com...",
    }}</dd>`;

```

Reply:
Yes Bu the `content` won't be a single node then so can't do `insertBefore` on this (Will figure this one out, currently gives:  `TypeError: Failed to execute 'insertBefore' on 'Node': parameter 1 is not of type 'Node'.`)